compiler/circle-mpqsolver/src/CircleMPQSolver.cpp

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "BisectionSolver.h"
+
+#include <arser/arser.h>
+#include <vconone/vconone.h>
+
+#include <luci/ImporterEx.h>
+#include <luci/Importer.h>
+#include <luci/CircleExporter.h>
+#include <luci/CircleFileExpContract.h>
+
+#include <iostream>
+#include <iomanip>
+#include <chrono>
+
+void print_version(void)
+{
+  std::cout << "circle-mpqsolver version " << vconone::get_string() << std::endl;
+  std::cout << vconone::get_copyright() << std::endl;
+}
+
+int entry(int argc, char **argv)
+{
+  const std::string bisection_str = "--bisection";
+
+  arser::Arser arser("circle-mpqsolver provides circle_model mixed precision quantization");
+
+  arser::Helper::add_version(arser, print_version);
+  arser::Helper::add_verbose(arser);
+
+  arser.add_argument("--data").required(true).help(".h5 file with test data");
+  arser.add_argument("--qerror_ratio")
+    .type(arser::DataType::FLOAT)
+    .default_value(0.5f)
+    .help("quantization error ratio");
+
+  arser.add_argument(bisection_str)
+    .nargs(1)
+    .type(arser::DataType::STR_VEC)
+    .help("Bisection method Q16 at input nodes"
+          "Default is 'auto'."
+          "Single optional argument: whether Q16 should be at input nodes "
+          "'auto', 'input', 'output'.");
```

Comment:
1. Seems to me that below codes check if the number of arguments is 1. But its type is set to `STR_VEC`. Maybe `STR`?
2. it says `Default is auto`. Then, `.default_value('auto')` might be needed.
3. Actually, it is very hard to know what values can be given -  'auto', 'true', 'false'.

Reply:
Ahhh... Ok. It was hardcoded, but in the wrong place, I'll fix it. Thank you.

Reply:
> 1. Seems to me that below codes check if the number of arguments is 1. But its type is set to `STR_VEC`. Maybe `STR`?

There may be other options, like .json file with frozen layers. But for now single option is just enough. Let it be STR. All other options may be set at .json file later as soon as there is need in them.